web/src/navigator.rs

```diff
@@ -43,61 +43,91 @@ impl NavigatorBackend for WebNavigatorBackend {
         vars_method: Option<(NavigationMethod, IndexMap<String, String>)>,
     ) {
         if let Some(window) = window() {
-            let url = if let Ok(parsed_url) = Url::parse(&url) {
-                self.pre_process_url(parsed_url).to_string()
-            } else {
-                url.to_string()
+            let document = match window.document() {
+                Some(document) => document,
+                None => return,
+            };
+            let body = match document.body() {
+                Some(body) => body,
+                None => return,
             };
 
-            //TODO: Should we return a result for failed opens? Does Flash care?
-            #[allow(unused_must_use)]
-            match (vars_method, window_spec) {
-                (Some((navmethod, formvars)), window_spec) => {
-                    let document = match window.document() {
-                        Some(document) => document,
-                        None => return,
-                    };
-
-                    let form = document
-                        .create_element("form")
-                        .unwrap()
-                        .dyn_into::<web_sys::HtmlFormElement>()
-                        .unwrap();
+            if url.trim().starts_with("javascript:") {
+                let target = window_spec.unwrap_or_else(|| "".to_string());
 
-                    form.set_attribute(
-                        "method",
-                        match navmethod {
-                            NavigationMethod::GET => "get",
-                            NavigationMethod::POST => "post",
-                        },
-                    );
+                if target.is_empty() || target == "_self" || target == "undefined" {
+                    let code = url.replacen("javascript:", "", 1);
 
-                    form.set_attribute("action", &url);
+                    let script = document.create_element("script").unwrap();
+                    script.set_inner_html(&code);
 
-                    if let Some(target) = window_spec {
-                        form.set_attribute("target", &target);
+                    let _append = body.append_child(&script);
+                    let _remove = body.remove_child(&script);
```

Comment:
I believe this should only be allowed if `allowScriptAccess` is true. We can pass `allow_script_access` to the NavigatorBackend when we make the player in `lib.rs`.

Also, I think we could move this to a separate `run_script` method (because it will possibly be used elsewhere).